### ISSUE TITLE
Add a way to turn Harfbuzz on and off

### ISSUE THREAD

**Issue by deepakjois** [deepakjois/luatex-harfbuzz#5]

**Comment by khaledhosny** [comment]
I think the simplest way is to make it a font option.

**Comment by deepakjois** [comment]
Yes, that might be a good option. I am still not entirely sure how font loading for this would work.

I could piggyback off luaotfload, in which case I can use their font database and font parsing spec as-is without inventing my own. If not, I will (unfortunately) have to create my own parallel code base that does all of it – parsing the font directive, parsing the font files, as well as maintaining a font cache/database for querying.

The problem is that luaotfload is just a thin wrapper over the fontloader in ConTeXt, and that fontloader is open source only in a notional sense. It has a very idiosyncratic coding style, and I barely understand it even after multiple readings. It doesn’t look possible to make contributions to it as well.

So I am strongly considering just writing all the code myself, packaging it cleanly as a rock and have some tests etc. Hopefully that will encourage others to contribute as well.

**Comment by khaledhosny** [comment]
I completely agree about luaotfload issues, and I also find its font loading support to be as problematic as its OpenType implementation; it doesn’t integrate with the system font loading (no support for FontConfig, Core Text or DirectWrite font loading APIs), it often does style linking differently than any one else (common source of bugs) and also slow. It also complicates things unnecessarily (like having a bunch of spec parsers providing different `\font` syntax for each, while in practice only one is used).

If it were me, I’d forget about luaotfload entirely, provide only a spec parser compatible with XeTeX (which is what macro packages like fontspec use), and have a binary module based on https://github.com/phst/lltxplatform for finding fonts and may be copy the style linking bits from XeTeX on top of that (since it works pretty well).

After that, we can consider the making harfbuzz optional part, e.g. if `harfbuzz=no` font option is given, detect if luaotfload is loaded and fallback to its `define_font` implementation.

**Comment by deepakjois** [comment]
I did not know about lltxplatform. I will take a look, and then look into adapting it.

**Comment by deepakjois** [comment]
Created [luafontmanager](https://github.com/deepakjois/luafontmanager). It seems to work on OS X for now, and I plan to extend support for Windows and Linux soon.
